src/plugins/charts/impl/XMLAComposer.ts

```diff
@@ -0,0 +1,106 @@
+import type {
+    AxisData,
+    Composer,
+    XMLASelector,
+} from "@/plugins/charts/widgets/api/ChartdataComposer";
+import { computed, type ComputedRef, type Ref, ref, watch } from "vue";
+
+import { useStore } from "@/composables/widgets/store";
+
+export class XMLAComposer implements Composer<XMLASelector> {
+    private selectorX: XMLASelector | undefined;
+    private selectorY: any = {};
+    private data: Ref<any> = ref({});
+    private store: IStore | undefined;
+    type: string = "XMLA";
+
+    addSelectorY(selector: XMLASelector, axisName: string) {
+        this.selectorY[axisName].push(selector);
+    }
+
+    getSelectorsY() {
+        return this.selectorY;
+    }
+
+    getSelectorY(axisName: string) {
+        if (!this.selectorY[axisName]) {
+            return [];
+        }
+
+        return this.selectorY[axisName];
+    }
+
+    setSelectorY(selector: XMLASelector, axisName) {
+        if (!this.selectorY[axisName]) {
+            this.selectorY[axisName] = [];
+        }
+
+        this.selectorY[axisName] = selector;
```

Comment:
should be a push not a replace

Reply:
Why it should be a push?

Doesn't it break a method namings?
addSmth -> push to state
setSmth -> replace state

Reply:
because this.selectorY[axisName] should be a array holding the selectors, if u set the selector to  this.selectorY[axisName] is not an array anymore, if want to replace iterate over and splice the elemnt